src/crypto/NoteEncryption.cpp

```diff
@@ -41,7 +41,7 @@ void KDF(unsigned char *K,
 }
 
 template<size_t MLEN>
-NoteEncryption<MLEN>::NoteEncryption(uint256 seed) : times_ran(0), seed(seed) {
+NoteEncryption<MLEN>::NoteEncryption(uint256 seed) : nonce(0), seed(seed) {
```

Comment:
The cached ``sk_enc`` would be state used purely for optimisation, which is different and usually less tricky than state relied on by an interface definition. But on reflection I'm ok with the nonce handling being defined this way.